### PR TITLE
Fix: Use `waitUntil` instead of `waitForLoadState` in page.goto()

### LINKED ISSUE #36698: E2E: Use `waitUntil` instead of `waitForLoadState` in `page.goto()`
In our E2E tests, there are `page.goto()`'s that use `waitForLoadState` as an option.

QsaNwwGWlg

We must change them to `waitUntil` because that's the valid option according to the [Playwright docs](https://playwright.dev/docs/api/class-page#page-goto-option-wait-until).

### PR DESCRIPTION
**Pull request by ravinderk**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

I found that in few playwright tests, we are using `waitForLoadState`. This is not correct argument to `goto` and `click`. I replace it with [`waitUntil`](https://playwright.dev/docs/api/class-frame#frame-goto).

Closes #36698 .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Changed playwright E2E tests should pass.

### CODE REVIEW

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/order-refund.spec.js [hunk 1]
```diff
@@ -82,7 +82,7 @@ test.describe.serial( 'WooCommerce Orders > Refund an order', () => {
 		// Do the refund
 		page.on( 'dialog', ( dialog ) => dialog.accept() );
 		await page.click( '.do-manual-refund', {
-			waitForLoadState: 'networkidle',
+			waitUntil: 'networkidle',
```
Comment (tammullen) [inline plugins/woocommerce/tests/e2e-pw/tests/merchant/order-refund.spec.js]: Thanks for looking at this one, it has drawn my attention to the documentation for `page.click` https://playwright.dev/docs/api/class-page#page-click which doesn't actually have either `waitForLoadState` or `waitUntil` as options.
I think this one needs looked at separately to the original issue of updating within `page.goto` so I think it is best for now to not make this change to `waitUntil` for this line. 
We will create a followup issue for this.
I also notice in general page.click is deprecated so we will also create a followup issue for this as well

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/order-refund.spec.js [hunk 2]
```diff
@@ -115,7 +115,7 @@ test.describe.serial( 'WooCommerce Orders > Refund an order', () => {
 		page.on( 'dialog', ( dialog ) => dialog.accept() );
 		await page.click( 'a.delete_refund', {
 			force: true,
-			waitForLoadState: 'networkidle',
+			waitUntil: 'networkidle',
```
Comment (tammullen) [inline plugins/woocommerce/tests/e2e-pw/tests/merchant/order-refund.spec.js]: Same as for the others within `page.click`s in this test I think we don't need to update yet until we figure out if they should even be within .click

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/order-refund.spec.js [hunk 3]
```diff
@@ -223,7 +223,7 @@ test.describe( 'WooCommerce Orders > Refund and restock an order item', () => {
 		await page.fill( '#refund_reason', 'No longer wanted' );
 		page.on( 'dialog', ( dialog ) => dialog.accept() );
 		await page.click( '.do-manual-refund', {
-			waitForLoadState: 'networkidle',
+			waitUntil: 'networkidle',
```
Comment (tammullen) [inline plugins/woocommerce/tests/e2e-pw/tests/merchant/order-refund.spec.js]: Same as for the others within `page.click`s in this test I think we don't need to update yet until we figure out if they should even be within .click

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/page-loads.spec.js [hunk 4]
```diff
@@ -83,7 +83,7 @@ for ( const currentPage of wcPages ) {
 
 					await page.click(
 						`li.wp-menu-open > ul.wp-submenu > li:has-text("${ currentPage.subpages[ i ].name }")`,
-						{ waitForLoadState: 'networkidle' }
+						{ waitUntil: 'networkidle' }
```
Comment (tammullen) [inline plugins/woocommerce/tests/e2e-pw/tests/merchant/page-loads.spec.js]: Same as for the others within page.clicks in this test I think we don't need to update yet until we figure out if they should even be within `page.click` options

**Comment by ravinderk** [comment]
@tammullen I reverted changes related to `page.click`. This pull request is ready to review.

**Review by lanej0 — DISMISSED** [review]
Looks good! Thank you for this contribution!

**Review by lanej0 — APPROVED** [review]

**Comment by nigeljamesstevenson** [comment]
Hi folks, a bit late to the party on this one.

I might be missing something here but it appears that the code `{ waitForLoadState: 'networkidle' }` has been removed rather than being replaced with `{ waitUntil: 'networkidle' }` as per the [playwright docs](https://playwright.dev/docs/navigations#custom-wait)

Is this expected @lanej0 @tammullen?

**Comment by lanej0** [comment]
That was me @nigeljamesstevenson -- the tests started failing with the update. I don't think we need an explicit wait in those places anyhow, so I tried removing and ran the tests a bunch of times without issue.

File: plugins/woocommerce/tests/e2e-pw/tests/admin-analytics/analytics-overview.spec.js [hunk 5]
```diff
@@ -6,8 +6,7 @@ test.describe( 'Analytics pages', () => {
 	test.afterEach( async ( { page } ) => {
 		// do some cleanup after each test to make sure things are where they should be
 		await page.goto(
-			'wp-admin/admin.php?page=wc-admin&path=%2Fanalytics%2Foverview',
-			{ waitForLoadState: 'networkidle' }
+			'wp-admin/admin.php?page=wc-admin&path=%2Fanalytics%2Foverview'
 		);
```
Comment (nigeljamesstevenson) [inline plugins/woocommerce/tests/e2e-pw/tests/admin-analytics/analytics-overview.spec.js:10]: `{ waitUntil: 'networkidle' }` has not been added

File: plugins/woocommerce/tests/e2e-pw/tests/admin-analytics/analytics-overview.spec.js [hunk 6]
```diff
@@ -52,8 +51,7 @@ test.describe( 'Analytics pages', () => {
 		// Create an array of the sections we're expecting to find.
 		const arrExpectedSections = [ 'Charts', 'Leaderboards', 'Performance' ];
 		await page.goto(
-			'wp-admin/admin.php?page=wc-admin&path=%2Fanalytics%2Foverview',
-			{ waitForLoadState: 'networkidle' }
+			'wp-admin/admin.php?page=wc-admin&path=%2Fanalytics%2Foverview'
 		);
```
Comment (nigeljamesstevenson) [inline plugins/woocommerce/tests/e2e-pw/tests/admin-analytics/analytics-overview.spec.js:55]: `{ waitUntil: 'networkidle' }` has not been added

**Pull request merged**